src/main/java/com/example/TaskFlow/services/UserValidationServiceImpl.java

```diff
@@ -1,47 +1,68 @@
 package com.example.TaskFlow.services;
 
+import com.example.TaskFlow.exceptions.UserCreateException;
+import com.example.TaskFlow.models.entities.User;
+import com.example.TaskFlow.repositories.UserRepository;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.TypedQuery;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.Optional;
+
 @Service
 @Slf4j
 public class UserValidationServiceImpl implements UserValidationService {
-@Override
-public  void emailValidation (String email) {
-    if (!email.trim().contains("@") || email.trim().length() < 4){
-        throw new RuntimeException("Email is not valid");
+
+    private final UserRepository userRepository;
+
+    public UserValidationServiceImpl(UserRepository userRepository) {
+        this.userRepository = userRepository;
     }
+
+    @Override
+public  void validateEmail (String email) {
+        Optional<User> user = userRepository.findUserByEmail(email);
+        if (email.isEmpty()){
+            throw  new UserCreateException("Please add an email");
+        }
+        if (user != null){
+            throw new UserCreateException("Email already used. Try again with another email!");
+        }
+
 }
 
     @Override
-    public void firstNameValidation(String firstName) throws IllegalArgumentException {
+    public void firstNameValidation(String firstName) throws UserCreateException {
```

Comment:
As suggested above, we should not have distinct validation methods for each property of the UserDTO. This comment is also valid for lastNameValidation. Also, please consider using the same naming pattern for your methods - use the keyword `validate` first for method naming.